www/main/components/addons/bani-controller/components/BaniController.jsx

```diff
@@ -0,0 +1,164 @@
+import { useStoreState, useStoreActions } from 'easy-peasy';
+import React, { useState, useEffect, useRef } from 'react';
+import PropTypes from 'prop-types';
+import { remote } from 'electron';
+import isOnline from 'is-online';
+
+import useSocketListeners from '../hooks/use-socket-listeners';
+import BaniControllerItem from './BaniControllerItem';
+import { Switch, Overlay } from '../../../../sttm-ui';
+
+import { getBaniControllerItems, generateQrCode, shareSync } from '../utils';
+
+const { tryConnection, onEnd } = shareSync;
+
+const { i18n } = remote.require('./app');
+
+const BaniController = ({ onScreenClose }) => {
+  const title = 'Mobile device sync';
+  const canvasRef = useRef(null);
+  // Local State
+  const [codeLabel, setCodeLabel] = useState('');
+  const [isFetchingCode, setFetchingCode] = useState(false);
+  const [isAdminPinVisible, setAdminPinVisibility] = useState(true);
+  // Store State
+  const { isListeners } = useStoreState(state => state.app);
+  const { adminPin, code, isConnected } = useStoreState(state => state.baniController);
+  const { setAdminPin, setCode, setConnection } = useStoreActions(
+    actions => actions.baniController,
+  );
+
+  const { overlayScreen } = useStoreState(state => state.app);
+  const { setOverlayScreen } = useStoreActions(actions => actions.app);
+  const { setListeners } = useStoreActions(actions => actions.app);
```

Comment:
btw just curious, why the two desctructuring assignments from the same actions.app?

Reply:
Good catch @AkalUstat 🔥